featurebyte/service/session_manager.py

```diff
@@ -64,18 +68,19 @@ async def get_feature_store_session(
         CredentialsError
             When the credentials used to access the feature store is missing or invalid
         """
-        user_to_use = self.user
-        if user_override is not None:
-            user_to_use = user_override
+        user_to_use = self.user if user_override is None else user_override
         try:
             if get_credential is not None:
-                credential = await get_credential(
-                    user_id=user_to_use.id, feature_store_name=feature_store.name
-                )
+                credential = await get_credential(user_to_use.id, feature_store.name)
             else:
-                credential = await self.credential_provider.get_credential(
+                credential = await self.credential_service.find(
                     user_id=user_to_use.id, feature_store_name=feature_store.name
                 )
```

Comment:
Could we remove the `find` method from `credential_service` and use `feature_store_service.get_document_as_dict` here?

Reply:
The find method was the previous get_credential method which contains additional logic than the normal get_document call. 
After this PR i'm looking to add the group and user_id lookup inside this method so i don't think the generic get_document function would work.